src/passport-saml/saml.ts

```diff
@@ -1554,6 +1556,13 @@ class SAML {
       metadata.EntityDescriptor.SPSSODescriptor.NameIDFormat = this.options.identifierFormat;
     }
 
+    if (this.options.wantAssertionsSigned) {
+      if (!this.options.cert) {
+        throw new Error('"cert" config parameter is required for signed assertions');
```

Comment:
I think this this option combination check should be at SAML's constructor/initialize method. Now this check is evaluated if and only if `generateServiceProviderMetadata` method is invoked. Earlier comments about allowing missing `cert` at all (regardless of value of `wantAssertionsSigned`) apply.

If this option combinarion check is moved to constructor/initialize method then you would not have to have this at `validatePostResponseAsync` method:
```javascript
      if (this.options.wantAssertionsSigned && !this.options.cert) {
        throw new Error("Invalid signature");
      }
  ```
because one would have had error during constructor invocation (if I interpreted javascript/typescript correctly...I'm not too familiar with these languages).

Reply:
Thank you for the constructive feedback.
Moving the option combination check might be an improvement indeed; I'll look into it.

Reply:
I've updated the code as you suggested.